addon/utils/plugins/table/tab-input-plugin.ts

```diff
@@ -24,11 +26,58 @@ export default class TableTabInputPlugin implements TabInputPlugin {
         allow: true,
         executor: this.tabHandler as unknown as ManipulationExecutor
       };
+    } else {
+      if(_manipulation.type === 'moveCursorToStartOfElement' || _manipulation.type === 'moveCursorAfterElement') {
+        if(this.isNextElementATable(selection)){
+          return {
+            allow: true,
+            executor: this.selectFirstCell,
+          }
+        }
+      } else if(_manipulation.type === 'moveCursorToEndOfElement' || _manipulation.type === 'moveCursorBeforeElement') {
+        if(this.isPreviousElementATable(selection)){
+          return {
+            allow: true,
+            executor: this.selectLastCell,
+          }
+        }
+      }
     }
     return null;
   }
 
-  tabHandler = (manipulation : Manipulation, editor: RawEditor) => {
+  isNextElementATable(selection: ModelSelection) {
+    const nextSibling = selection.anchor?.nodeAfter() as ModelElement;
+    if(!nextSibling) return false;
```

Comment:
I think this is more readable if converted to
```
if (nextSibling) {
    return nextSibling.type === 'table';
}
else {
  return false;
}
```

Reply:
what about
`return nextSibling?.type === "table";`

Reply:
I'd say the if/else is more clear if quickly eyeing the code, but I'm OK with both.